src/os/osexec.go

```diff
@@ -0,0 +1,56 @@
+//go:build linux && !baremetal && !darwin && !tinygo.wasm && !arm64
+
+// arm64 does not have a fork syscall, so ignore it for now
+// TODO: add support for arm64 with clone or use musl implementation
+
+package os
+
+import (
+	"syscall"
+	"unsafe"
+)
+
+func fork() (pid int, err error) {
+	ret, _, err := syscall.Syscall(syscall.SYS_FORK, 0, 0, 0)
+	if int(ret) != 0 {
+		errno := err.(syscall.Errno)
+		if int(errno) != 0 {
+			return -1, errno
+		}
+	}
+	return int(ret), nil
+}
```

Comment:
Can you please either implement `fork()` for arm64, or just use the libc function?

For example (this code is entirely untested but something like that should work):

```suggestion
func fork() (pid int32, err error) {
	pid = libc_fork()
	if pid != 0 {
		err = syscall.Errno(*libc_errno())
	}
	return
}

//export fork
func libc_fork() int32

// Internal musl function to get the C errno pointer.
//export __errno_location
func libc_errno() *int32
```

Reply:
Thanks for the input, this seems to be a better idea. I had a look at the execve implementation, but I think there it does not make sense to call into musl since it is the exact same thing we do here:

```c
int execve(const char *path, char *const argv[], char *const envp[])
{
	return syscall(SYS_execve, path, argv, envp);
}

```